### ISSUE TITLE
StarDist model loading

### ISSUE THREAD

**Issue by pwalczysko** [ome/omero-guide-python#55]
When executing the https://github.com/ome/omero-guide-python/blob/master/notebooks/idr0062_prediction_save.ipynb on master using conda, and exchanging the kernel as instructed, after executing the cell 

```
from stardist.models import StarDist2D
model_versatile = StarDist2D.from_pretrained('2D_demo')
```

I got 

```
Found model '2D_demo' for 'StarDist2D'.

2023-03-16 15:17:24.226280: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE4.1 SSE4.2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[1], line 2
      1 from stardist.models import StarDist2D
----> 2 model_versatile = StarDist2D.from_pretrained('2D_demo')

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/csbdeep/models/base_model.py:79, in BaseModel.from_pretrained(cls, name_or_alias)
     77 try:
     78     get_model_details(cls, name_or_alias, verbose=True)
---> 79     return get_model_instance(cls, name_or_alias)
     80 except ValueError as e:
     81     if name_or_alias is not None:

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/csbdeep/models/pretrained.py:107, in get_model_instance(cls, key_or_alias)
    105 def get_model_instance(cls, key_or_alias):
    106     path = get_model_folder(cls, key_or_alias)
--> 107     model = cls(config=None, name=path.stem, basedir=path.parent)
    108     model.basedir = None # make read-only
    109     return model

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/stardist/models/model2d.py:292, in StarDist2D.__init__(self, config, name, basedir)
    290 def __init__(self, config=Config2D(), name=None, basedir='.'):
    291     """See class docstring."""
--> 292     super().__init__(config, name=name, basedir=basedir)

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/stardist/models/base.py:222, in StarDistBase.__init__(self, config, name, basedir)
    221 def __init__(self, config, name=None, basedir='.'):
--> 222     super().__init__(config=config, name=name, basedir=basedir)
    223     threshs = dict(prob=None, nms=None)
    224     if basedir is not None:

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/csbdeep/models/base_model.py:111, in BaseModel.__init__(self, config, name, basedir)
    109     self._update_and_check_config()
    110 self._model_prepared = False
--> 111 self.keras_model = self._build()
    112 if config is None:
    113     self._find_and_load_weights()

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/stardist/models/model2d.py:308, in StarDist2D._build(self)
    306     pooled *= pool
    307     for _ in range(self.config.unet_n_conv_per_depth):
--> 308         pooled_img = Conv2D(self.config.unet_n_filter_base, self.config.unet_kernel_size,
    309                             padding='same', activation=self.config.unet_activation)(pooled_img)
    310     pooled_img = MaxPooling2D(pool)(pooled_img)
    312 unet_base = unet_block(**unet_kwargs)(pooled_img)

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/keras/engine/base_layer.py:976, in Layer.__call__(self, *args, **kwargs)
    970 # Functional Model construction mode is invoked when `Layer`s are called on
    971 # symbolic `KerasTensor`s, i.e.:
    972 # >> inputs = tf.keras.Input(10)
    973 # >> outputs = MyLayer()(inputs)  # Functional construction mode.
    974 # >> model = tf.keras.Model(inputs, outputs)
    975 if _in_functional_construction_mode(self, inputs, args, kwargs, input_list):
--> 976   return self._functional_construction_call(inputs, args, kwargs,
    977                                             input_list)
    979 # Maintains info about the `Layer.call` stack.
    980 call_context = base_layer_utils.call_context()

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/keras/engine/base_layer.py:1114, in Layer._functional_construction_call(self, inputs, args, kwargs, input_list)
   1109     training_arg_passed_by_framework = True
   1111 with call_context.enter(
   1112     layer=self, inputs=inputs, build_graph=True, training=training_value):
   1113   # Check input assumptions set after layer building, e.g. input shape.
-> 1114   outputs = self._keras_tensor_symbolic_call(
   1115       inputs, input_masks, args, kwargs)
   1117   if outputs is None:
   1118     raise ValueError('A layer\'s `call` method should return a '
   1119                      'Tensor or a list of Tensors, not None '
   1120                      '(layer: ' + self.name + ').')

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/keras/engine/base_layer.py:848, in Layer._keras_tensor_symbolic_call(self, inputs, input_masks, args, kwargs)
    846   return tf.nest.map_structure(keras_tensor.KerasTensor, output_signature)
    847 else:
--> 848   return self._infer_output_signature(inputs, args, kwargs, input_masks)

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/keras/engine/base_layer.py:886, in Layer._infer_output_signature(self, inputs, args, kwargs, input_masks)
    880 with backend.name_scope(self._name_scope()):  # pylint: disable=not-callable
    881   with autocast_variable.enable_auto_cast_variables(
    882       self._compute_dtype_object):
    883     # Build layer if applicable (if the `build` method has been
    884     # overridden).
    885     # TODO(kaftan): do we maybe_build here, or have we already done it?
--> 886     self._maybe_build(inputs)
    887     inputs = self._maybe_cast_inputs(inputs)
    888     outputs = call_fn(inputs, *args, **kwargs)

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/keras/engine/base_layer.py:2659, in Layer._maybe_build(self, inputs)
   2654 if not hasattr(self.build, '_is_default'):
   2655   # Any setup work performed only once should happen in an `init_scope`
   2656   # to avoid creating symbolic Tensors that will later pollute any eager
   2657   # operations.
   2658   with tf_utils.maybe_init_scope(self):
-> 2659     self.build(input_shapes)  # pylint:disable=not-callable
   2660 # We must set also ensure that the layer is marked as built, and the build
   2661 # shape is stored since user defined build functions may not be calling
   2662 # `super.build()`
   2663 Layer.build(self, input_shapes)

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/keras/layers/convolutional.py:197, in Conv.build(self, input_shape)
    189   raise ValueError(
    190       'The number of input channels must be evenly divisible by the number '
    191       'of groups. Received groups={}, but the input has {} channels '
    192       '(full input shape is {}).'.format(self.groups, input_channel,
    193                                          input_shape))
    194 kernel_shape = self.kernel_size + (input_channel // self.groups,
    195                                    self.filters)
--> 197 self.kernel = self.add_weight(
    198     name='kernel',
    199     shape=kernel_shape,
    200     initializer=self.kernel_initializer,
    201     regularizer=self.kernel_regularizer,
    202     constraint=self.kernel_constraint,
    203     trainable=True,
    204     dtype=self.dtype)
    205 if self.use_bias:
    206   self.bias = self.add_weight(
    207       name='bias',
    208       shape=(self.filters,),
   (...)
    212       trainable=True,
    213       dtype=self.dtype)

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/keras/engine/base_layer.py:647, in Layer.add_weight(self, name, shape, dtype, initializer, regularizer, trainable, constraint, use_resource, synchronization, aggregation, **kwargs)
    642     tf_logging.warning(
    643         '`caching_device` does not work with mixed precision API. Ignoring '
    644         'user specified `caching_device`.')
    645     caching_device = None
--> 647 variable = self._add_variable_with_custom_getter(
    648     name=name,
    649     shape=shape,
    650     # TODO(allenl): a `make_variable` equivalent should be added as a
    651     # `Trackable` method.
    652     getter=getter,
    653     # Manage errors in Layer rather than Trackable.
    654     overwrite=True,
    655     initializer=initializer,
    656     dtype=dtype,
    657     constraint=constraint,
    658     trainable=trainable,
    659     use_resource=use_resource,
    660     collections=collections_arg,
    661     synchronization=synchronization,
    662     aggregation=aggregation,
    663     caching_device=caching_device)
    664 if regularizer is not None:
    665   # TODO(fchollet): in the future, this should be handled at the
    666   # level of variable creation, and weight regularization losses
    667   # should be variable attributes.
    668   name_in_scope = variable.name[:variable.name.find(':')]

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/tensorflow/python/training/tracking/base.py:813, in Trackable._add_variable_with_custom_getter(self, name, shape, dtype, initializer, getter, overwrite, **kwargs_for_getter)
    803   if (checkpoint_initializer is not None and
    804       not (isinstance(initializer, CheckpointInitialValueCallable) and
    805            (initializer.restore_uid > checkpoint_initializer.restore_uid))):
   (...)
    810     # then we'll catch that when we call _track_trackable. So this is
    811     # "best effort" to set the initializer with the highest restore UID.
    812     initializer = checkpoint_initializer
--> 813 new_variable = getter(
    814     name=name,
    815     shape=shape,
    816     dtype=dtype,
    817     initializer=initializer,
    818     **kwargs_for_getter)
    820 # If we set an initializer and the variable processed it, tracking will not
    821 # assign again. It will add this variable to our dependencies, and if there
    822 # is a non-trivial restoration queued, it will handle that. This also
    823 # handles slot variables.
    824 if not overwrite or isinstance(new_variable, Trackable):

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/keras/engine/base_layer_utils.py:117, in make_variable(name, shape, dtype, initializer, trainable, caching_device, validate_shape, constraint, use_resource, collections, synchronization, aggregation, partitioner)
    112   use_resource = True
    113 # In theory, in `use_resource` is True and `collections` is empty
    114 # (that is to say, in TF2), we can use tf.Variable.
    115 # However, this breaks legacy (Estimator) checkpoints
    116 # because it changes variable names. Remove this when V1 is fully deprecated.
--> 117 return tf.compat.v1.Variable(
    118     initial_value=init_val,
    119     name=name,
    120     trainable=trainable,
    121     caching_device=caching_device,
    122     dtype=variable_dtype,
    123     validate_shape=validate_shape,
    124     constraint=constraint,
    125     use_resource=use_resource,
    126     collections=collections,
    127     synchronization=synchronization,
    128     aggregation=aggregation,
    129     shape=variable_shape if variable_shape else None)

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/tensorflow/python/ops/variables.py:266, in VariableMetaclass.__call__(cls, *args, **kwargs)
    264 def __call__(cls, *args, **kwargs):
    265   if cls is VariableV1:
--> 266     return cls._variable_v1_call(*args, **kwargs)
    267   elif cls is Variable:
    268     return cls._variable_v2_call(*args, **kwargs)

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/tensorflow/python/ops/variables.py:212, in VariableMetaclass._variable_v1_call(cls, initial_value, trainable, collections, validate_shape, caching_device, name, variable_def, dtype, expected_shape, import_scope, constraint, use_resource, synchronization, aggregation, shape)
    210 if aggregation is None:
    211   aggregation = VariableAggregation.NONE
--> 212 return previous_getter(
    213     initial_value=initial_value,
    214     trainable=trainable,
    215     collections=collections,
    216     validate_shape=validate_shape,
    217     caching_device=caching_device,
    218     name=name,
    219     variable_def=variable_def,
    220     dtype=dtype,
    221     expected_shape=expected_shape,
    222     import_scope=import_scope,
    223     constraint=constraint,
    224     use_resource=use_resource,
    225     synchronization=synchronization,
    226     aggregation=aggregation,
    227     shape=shape)

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/tensorflow/python/ops/variables.py:205, in VariableMetaclass._variable_v1_call.<locals>.<lambda>(**kwargs)
    188 def _variable_v1_call(cls,
    189                       initial_value=None,
    190                       trainable=None,
   (...)
    202                       aggregation=VariableAggregation.NONE,
    203                       shape=None):
    204   """Call on Variable class. Useful to force the signature."""
--> 205   previous_getter = lambda **kwargs: default_variable_creator(None, **kwargs)
    206   for _, getter in ops.get_default_graph()._variable_creator_stack:  # pylint: disable=protected-access
    207     previous_getter = _make_getter(getter, previous_getter)

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/tensorflow/python/ops/variable_scope.py:2612, in default_variable_creator(next_creator, **kwargs)
   2610 if use_resource:
   2611   distribute_strategy = kwargs.get("distribute_strategy", None)
-> 2612   return resource_variable_ops.ResourceVariable(
   2613       initial_value=initial_value,
   2614       trainable=trainable,
   2615       collections=collections,
   2616       validate_shape=validate_shape,
   2617       caching_device=caching_device,
   2618       name=name,
   2619       dtype=dtype,
   2620       constraint=constraint,
   2621       variable_def=variable_def,
   2622       import_scope=import_scope,
   2623       distribute_strategy=distribute_strategy,
   2624       synchronization=synchronization,
   2625       aggregation=aggregation,
   2626       shape=shape)
   2627 else:
   2628   return variables.RefVariable(
   2629       initial_value=initial_value,
   2630       trainable=trainable,
   (...)
   2641       aggregation=aggregation,
   2642       shape=shape)

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/tensorflow/python/ops/variables.py:270, in VariableMetaclass.__call__(cls, *args, **kwargs)
    268   return cls._variable_v2_call(*args, **kwargs)
    269 else:
--> 270   return super(VariableMetaclass, cls).__call__(*args, **kwargs)

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/tensorflow/python/ops/resource_variable_ops.py:1602, in ResourceVariable.__init__(self, initial_value, trainable, collections, validate_shape, caching_device, name, dtype, variable_def, import_scope, constraint, distribute_strategy, synchronization, aggregation, shape)
   1600   self._init_from_proto(variable_def, import_scope=import_scope)
   1601 else:
-> 1602   self._init_from_args(
   1603       initial_value=initial_value,
   1604       trainable=trainable,
   1605       collections=collections,
   1606       caching_device=caching_device,
   1607       name=name,
   1608       dtype=dtype,
   1609       constraint=constraint,
   1610       synchronization=synchronization,
   1611       aggregation=aggregation,
   1612       shape=shape,
   1613       distribute_strategy=distribute_strategy)

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/tensorflow/python/ops/resource_variable_ops.py:1756, in ResourceVariable._init_from_args(self, initial_value, trainable, collections, caching_device, name, dtype, constraint, synchronization, aggregation, distribute_strategy, shape)
   1754   else:
   1755     shape = initial_value.shape
-> 1756   handle = eager_safe_variable_handle(
   1757       initial_value=initial_value,
   1758       shape=shape,
   1759       shared_name=shared_name,
   1760       name=name,
   1761       graph_mode=self._in_graph_mode)
   1762 # pylint: disable=protected-access
   1763 if (self._in_graph_mode and initial_value is not None and
   1764     initial_value.op._get_control_flow_context() is not None):

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/tensorflow/python/ops/resource_variable_ops.py:238, in eager_safe_variable_handle(initial_value, shape, shared_name, name, graph_mode)
    196 """Creates a variable handle with information to do shape inference.
    197 
    198 The dtype is read from `initial_value` and stored in the returned
   (...)
    235   The handle, a `Tensor` of type `resource`.
    236 """
    237 dtype = initial_value.dtype.base_dtype
--> 238 return _variable_handle_from_shape_and_dtype(shape, dtype, shared_name, name,
    239                                              graph_mode, initial_value)

File ~/opt/anaconda3/envs/guide-python/lib/python3.9/site-packages/tensorflow/python/ops/resource_variable_ops.py:178, in _variable_handle_from_shape_and_dtype(shape, dtype, shared_name, name, graph_mode, initial_value)
    175 handle_data = cpp_shape_inference_pb2.CppShapeInferenceResult.HandleData()
    176 handle_data.is_set = True
    177 handle_data.shape_and_type.append(
--> 178     cpp_shape_inference_pb2.CppShapeInferenceResult.HandleShapeAndType(
    179         shape=shape.as_proto(), dtype=dtype.as_datatype_enum))
    181 if initial_value is not None and initial_value.dtype == dtypes.variant:
    182   extra_handle_data = get_eager_safe_handle_data(initial_value)

TypeError: Parameter to MergeFrom() must be instance of same class: expected tensorflow.TensorShapeProto got tensorflow.TensorShapeProto.

```

**Comment by pwalczysko** [comment]
This works fine using Google Colab

**Comment by jburel** [comment]
Replaced stardist by cellpose in #61
